runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.h

```diff
@@ -43,50 +36,155 @@ void configureHeader();
 void insertNoOpCommand(unsigned int numPadBytes);
 }
 
-#define OSTREAM_OP(O_TYPE, TYPE) O_TYPE &operator<<(O_TYPE &os, const TYPE &s);
-
 namespace mlir::iree_compiler::AMDAIE {
-#define TO_STRING(TYPE) std::string to_string(const TYPE &t);
+struct TileLoc {
+  int col, row;
+  inline bool operator<(const TileLoc& rhs) const {
+    return std::tie(col, row) < std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator==(const TileLoc& rhs) const {
+    return std::tie(col, row) == std::tie(rhs.col, rhs.row);
+  }
+
+  bool operator!=(const TileLoc& rhs) const { return !(*this == rhs); }
+
+  operator XAie_LocType() const { return XAie_TileLoc(col, row); }
+  TileLoc(XAie_LocType loc) : col(loc.Col), row(loc.Row) {}
+  TileLoc(int col, int row) : col(col), row(row) {}
+};
+
+enum class AMDAIEDevice : uint32_t {
+  xcvc1902 = 1,
+  xcve2302 = 2,
+  xcve2802 = 3,
+  npu = 4,
```

Comment:
Having 2 enums with the same value is strange, can we just remove one of them?

Reply:
`mlir-aie` has `npu1` which is ... strange for an enum that signifies "the whole device", hence just `npu`. The intent is to remove `npu1_*` when we ditch the old passes that reference those enum variants.